lib/gtfs/direction.ex

```diff
@@ -5,7 +5,7 @@ defmodule Gtfs.Direction do
 
   @type t :: %__MODULE__{
           route_id: Route.id(),
-          direction_id: id(),
+          direction_id: id() | nil,
```

Comment:
When would this be `nil`?

Reply:
It wouldn't in practice, but this value comes from `id_from_string/1`, which (for other inputs) might return a nil so dialyzer wants to see one here too.

Reply:
Looks like it's just [this use](https://github.com/mbta/skate/pull/67/files#diff-559bd7d15af65384238b8072a36fafdeR45) in `concentrate/parser/swiftly_realtime_vehicles.ex` that needs the `nil` check in `id_from_string`:
```elixir
def decode_vehicle(vehicle_data) do
  VehiclePosition.new(
    direction_id: vehicle_data |> Map.get("directionId") |> Direction.id_from_string(),
```

If the nil check is handled in that function before the call to `id_from_string` instead of adding a `nil` case to `id_from_string`, I think you could avoid needing `id_from_string` to return nil, so this wouldn't be needed.